### DISCUSSION TITLE
Desculpe, estava tentando colocar uma formação na célula, ate tentei colocar o código no componente.

### DISCUSSION THREAD

**Discussion by oilujrasec** [Daniel09Fernandes/Component_OpenOffice#71] [General]
_linha := 2;
      QryRecolhimentoRateio.First;
      while not QryRecolhimentoRateio.Eof do
      begin
        OpenofficeCalc.SetValue(_linha, 'A', FormatFloat('000000', QryRecolhimentoRateioCAD_MATRICULA.AsInteger));
        OpenofficeCalc.SetValue(_linha, 'B', QryRecolhimentoRateioCAD_NOME.AsString);
        OpenofficeCalc.SetValue(_linha, 'C', QryRecolhimentoRateioRCR_REMUNERACAO_SEM_DECIMO.AsFloat, ftNumeric);
        OpenofficeCalc.SetValue(_linha, 'D', QryRecolhimentoRateioRCR_DECIMO.AsFloat, ftNumeric);

        **OpenofficeCalc.setFormat(_linha, 'D', '#,##0.00');**
        {Como fazer isso?}

        Inc(_linha);
        QryRecolhimentoRateio.Next;
      end;

**Comment by Daniel09Fernandes** [comment]
Olá @oilujrasec  subi uma alteração, para poder formatar os numeros,
Voce pode fazer assim: 

![image](https://github.com/user-attachments/assets/57749c75-27de-42f6-8561-2875a1d58adf)

Estes são os formatos 
 //0; //1234,2
  //1; //1234
  //3; //1.234
  //4; //1.234,20
  //10; //123420%
  //11; //123420,00%
  //20; //R$ 1.234
  //21; //R$ 1.234,20
  //24; // 1.234,20 BRL
  //30; // DD/MM/YY
  //31; // DDDD DD/MM/YY
  //32; // MM/YY
  //33; // DD/MMMM
  //34; // MMMM
  //35; // TRIMESTRE
  //36; // DD/MM/YYYY
  //39; // DD MMMM YY
  //40; // DD MMMM YYYY
 
 TNumberMask = (Default, NoDecimalSeparator, WithDecimalSeparator = 3, WithDecimalSeparatorAndComma, Percent = 10, PercentWithComma, CurrencySymbolWithoutDecimal = 20,
                   CurrencySymbolWithDecimal, CurrencySuffix = 24, Date_dd_mm_yy = 30, Date_dddd_dd_mm_yyy,  Date_mm_yy, Date_dd_mmmm, Date_mmmm,
                   Date_QUARTER, Date_Default, Date_dd_mmmm_yy = 39, Date_dd_mmmm_yyyy);
